views/attendance/attendance_form.php

```diff
@@ -0,0 +1,692 @@
+        $changed_zip = $_POST['zip-input-change'];
+
+        $pageInformation[$selected_edit_num]['numChildren'] = $changed_num_children;
+        $pageInformation[$selected_edit_num]['zip'] = $changed_zip;
+
+        $_SESSION['serializedInfo'] = serializeParticipantMatrix($pageInformation);
+    }
+    else if(isset($_POST['lookupId'])){
+        $lookupId = $_POST['pidLookup'];
+
+        //lookup most recent info from DB --either from class info or from intake packet
+        $resultClassPastAttendance = $db->no_param_query(
+                "select * from classattendancedetails " .
+                "where participantid = {$lookupId} " .
+                "order by date desc; "
+        );
+
+        $fn = $mi = $ln = $dob = $race = $zip = $nc = $isNew = null;
+        //found past attendance history
+        if(pg_num_rows($resultClassPastAttendance)){
+            $row = pg_fetch_assoc($resultClassPastAttendance); //most recent row
+            $fn = $row['firstname'];
+            $mi = $row['middleinit'];
+            $ln = $row['lastname'];
+            $dob = $row['dateofbirth'];
+            $race = $row['race'];
+            $zip = $row['zipcode'];
+            $nc = $row['numchildren'];
+            $isNew = false;
+        } else{ //we didn't find that information find name and other info
+            //grab info from intake packet
+            $resultPersonLookup = $db->query(
+                "SELECT pe.firstname, pe.middleinit, pe.lastname, pa.dateofbirth, pa.race " .
+                "FROM people pe, participants pa " .
+                "WHERE pe.peopleid = pa.participantid " .
+                "AND pe.peopleid = $1;", [$lookupId]);
+            $row = pg_fetch_assoc($resultPersonLookup);
+            $fn = $row['firstname'];
+            $mi = $row['middleinit'];
+            $ln = $row['lastname'];
+            $dob = $row['dateofbirth'];
+            $race = $row['race'];
+            $isNew = true;
+        }
+
+        //look for duplicates
+        for($k = 0; $k < count($pageInformation); $k++){
+            if($pageInformation[$k]['pid'] == $lookupId){
+                $duplicatePerson = true;
+                $successAddingPerson = false;
+            }
+        }
+
+        //add person to pageInformation
+        if(!$duplicatePerson){
+            $pageInformation[] = array(
+                "pid"           => $lookupId,
+                "fn"            => $fn,
+                "mi"            => $mi,
+                "ln"            => $ln,
+                "dob"           => $dob,
+                "zip"           => $zip,
+                "numChildren"   => $nc,
+                "race"          => $race,
+                "comments"      => null,
+                "present"       => true,
+                "isNew"         => $isNew, //isNew field from DB
+                //people who haven't completed the intake forms and just filled out info in the "no intake form" section
```

Comment:
I don't think there is a `firstClass` . `isNew` is in ParticipantClassAttendance and could happen before (edge case) and after Intake packets as far as I know.

Reply:
firstClass is used to determine if a new record is going to be created for the person. When the form data is submitted, if firstclass is true, then a new person/participant is created in the system. If it is false, then that means we already found an id in the system for them.

Reply:
Thanks for the clarification!